### PR TITLE
YAML based config

### PR DESCRIPTION
**Pull request by avishniakov**
I reworked the template to use the YAML configuration instead of the Python file. This makes this template reusable for a broader count of use cases and makes it aligned with general ZenML configuration best practices.

- Configuration for HP tuning and other uses is now passed as `extra` config via YAML
- Search space objects support lists or ranges. For range a dictionary of `{"range":{"start":1,"end":10,"step":2}}` is used.
- Artifacts folder and inner objects are no longer needed
- New `get_model_from_config` step to reconstruct ClassifierMixin from string-based configurations

Depends on https://github.com/zenml-io/zenml/pull/1876

### CODE REVIEW

**Comment by fa9r** [comment]
Not really familiar with config best practices so I'm tagging @bcdurak instead who worked on this before I believe

**Comment by avishniakov** [comment]
MacOS tests are broken, to be followed up [here](https://zenml.atlassian.net/browse/OSS-2514)

File: template/config.yaml
Comment (strickvl) [inline template/config.yaml:1]: Nothing to do with this, but while I can see that people might prefer this kind of config for simple configurations, but I find this a decent bit less clear than having a Python-based config.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -35,4 +35,4 @@ jobs:
         with:
           stack-name: ${{ matrix.stack-name }}
           python-version: ${{ matrix.python-version }}
-          ref-zenml: develop
+          ref-zenml: feature/OSS-2494-evaluate-config-early
```
[the remark this answers is not in the document]
Comment (avishniakov) [reply .github/workflows/ci.yml:39]: This is fine, I update this ref every time I do some development here, and not always `develop` is the branch with necessary changes in it.
I will change that to develop once PR on the core side is merged.

**Review by strickvl — APPROVED** [review]

File: template/config.yaml
Comment (avishniakov) [reply template/config.yaml:1]: Yeah, I was also preferring python, but it is breaking caching logic currently and might not lead to new pipeline version generated in ZenML.

**Pull request merged**